test/compilable/ice21696.d

```diff
@@ -0,0 +1,15 @@
+module ice21696;
+
+// https://issues.dlang.org/show_bug.cgi?id=21696
+
+void func(double[1][1] stuff = [[1.0]])
+{
+    bool myFunc()
+    {
+        if (stuff[])
+            return true;
+        return false;
+    }
+
+    myFunc();
+}
```

Comment:
The test needs to do more than compile. We don't know if the code generator is actually generating useful code. Need to test both returns of myFunc().

Reply:
done.